eodatasets3/images.py

```diff
@@ -878,6 +936,147 @@ def create_thumbnail_singleband(
                         tmpdataset.write(out_data[i])
                 self.create_thumbnail(temp_files, out_file, static_stretch=stretch)
 
+    def create_thumbnail_singleband_from_numpy(
+        self,
+        input_data: numpy.array,
+        bit: int = None,
+        lookup_table: Dict[int, Tuple[int, int, int]] = None,
+        input_geobox: GridSpec = None,
+        nodata: int = -999,
+    ) -> bytes:
+        """
+        Output a thumbnail ready bytes from the input numpy array.
+        This takes a valid raster data (numpy arrary) and return
+        out bytes with only the values of the bit (integer) as white.
+        """
+        if bit is not None and lookup_table is not None:
+            raise ValueError(
+                "Please set either bit or lookup_table, and not both of them"
+            )
+        if bit is None and lookup_table is None:
+            raise ValueError(
+                "Please set either bit or lookup_table, you haven't set either of them"
+            )
+
+        # the output_data has three layers, as R, G, B
+        out_data, stretch = self.filter_singleband_data(input_data, bit, lookup_table)
+        return self.create_thumbnail_from_numpy(
+            rgb=out_data,
+            static_stretch=stretch,
+            input_geobox=input_geobox,
+            nodata=nodata,
+        )
+
+    def filter_singleband_data(
```

Comment:
I like that you removed the duplication here :+1: 

I'd leave this method as private, though, to keep our public footing small.

```suggestion
    def _filter_singleband_data(
```

Reply:
Make sense to me.